### DISCUSSION TITLE
Is there a way to selectively record bodies?

### DISCUSSION THREAD

**Discussion by cadop** [NVIDIA/warp#18] [Q&A] [answered]
Using, for example, the example sim_rigid_fem, is there a way to record only the sphere and not the beam?  Likewise, could there be a reduced version of the beam recorded?  At high meshes and timesteps, the USD file for that scene quickly hits over 60gb, I assume from all the particles being recorded.

**Comment by jlinBE** [comment]
You can get the coordinates and velocities out of the state objects (e.g. state_0, state_1) as arrays.  Take a look at the State and Model classes in warp\sim\model.py.

For example:
body_coordinates = state_1.body_q.numpy()

**Reply by cadop** [reply]
Thanks for the help.  I guess I didn't explain very clearly, but I had figured this out!

**Comment by cadop** [comment, marked as answer]
The best way I could find is to render based on object types (although I think this can easily extend to body ids). 

The `render.py` in sim is subclassing the main renderer, so I added a few extra parameters.  The heart of the modifications is:

```

    def render(self, state: warp.sim.State, particle_size=0.1, particles=True, meshes=True, bodies=True):

        if (self.model.particle_count):

            particle_q = state.particle_q.numpy()

            # render particles
            if particles: self.render_points("particles", particle_q, radius=particle_size)

            # render tris
            if meshes:
                if (self.model.tri_count):
                    self.render_mesh("surface", particle_q, self.model.tri_indices.numpy().squeeze())

            # render springs
            if (self.model.spring_count):
                self.render_line_list("springs", particle_q, self.model.spring_indices.numpy().squeeze(), [], 0.1)
```
